cmd/endpoint.go

```diff
@@ -744,3 +745,36 @@ func updateDomainIPs(endPoints set.StringSet) {
 		return !(strings.HasPrefix(ip, "127.") || strings.HasPrefix(ip, "::1") || strings.HasPrefix(ip, "[::1]"))
 	}, "")
 }
+
+// localEndpointsCPUHardware - returns ServerCPUHardwareInfo only for the
+// local endpoints from given list of endpoints
+func localEndpointsCPUHardware(endpoints EndpointList, r *http.Request) ServerCPUHardwareInfo {
+
+	var cpuHardwares []cpuinfo.InfoStat
+	scratchSpace := map[string]bool{}
+	for _, endpoint := range endpoints {
+		// Only proceed for local endpoints
+		if endpoint.IsLocal {
```

Comment:
Keep happy path to the left:

```suggestion
		if !endpoint.IsLocal {
			continue
		}
```

Reply:
:+1:  Good point to have.